### DISCUSSION TITLE
Best practices for enabling Vendor on an existing Seller app

### DISCUSSION THREAD

**Discussion by phstc** [amzn/selling-partner-api-models#4288] [General]
I want to expand my Seller app to include Vendor access. I’m considering whether to enable Vendor access on my existing app or create a new app entirely.

Questions:

- Authorization flow: Are there significant differences in the authorization flow between Seller and Vendor access? Specifically, how can I ensure that Seller users don’t see Vendor references and vice versa?
- Risks and benefits: What are the potential risks or benefits of enabling Vendor access on an existing app versus creating a new app?

Any other considerations I should keep in mind?

**Comment by puppsupr** [comment]
Hi @phstc,
Vendor have only [Website Authorization](https://developer-docs.amazon.com/sp-api/docs/website-authorization-workflow). Also when vendors authorize the app, 3P developer gets back a vendor group id [selling_partner_id] => [amzn1.vg](http://amzn1.vg/).XXXXXX from Amazon.
For what business use-cases you are looking to extend the app to vendors ?

**Reply by phstc** [reply]
Hi @puppsupr 

Thank you for your answer! We work with 3P, and we are expanding to 1P. 

For 1P, for now, we want to retrieve invoices and purchase orders..

We are considering having the same application (our application, not the Amazon app specifically) for both types of sellers, letting them add a vendor or 3P connections. 

We were unsure what the best approach would be—two or one app. Our application will be the same. The same account (under our system) will be able to add as many connections (1P or 3P) as they want. Then, there will be sections for each one on our application. 

It seems we should use the same Amazon app we already have. However, I have concerns about whether it can cause issues supporting both.

> For vendor applications, you can replace the Seller Central URL with a Vendor Central URL.

It seems the change is only with the auth URLs, and the vendor group ID you've mentioned.

How is the vendor group ID used? Is it similar to the Seller Partner ID?

**Reply by phstc** [reply]
Hey @puppsupr, following up on this. I would love to hear your recommendation.

**Reply by phstc** [reply]
Adding you two in the the loop since I saw you replying vendor questions 🙏  @MeenaAmz @ritikacd

**Reply by ritikacd** [reply]
Hi @phstc 

That's right. Vendor group IDs act as identifiers for vendors just like the merchant token for sellers. For vendor authorizations, you will receive a vendor code in the same `selling_partner_id` parameter. Please find more details [here](https://developer-docs.amazon.com/sp-api/docs/selling-partner-appstore-authorization-workflow#authorization-uri-details) about constructing an authorization URI for the vendors. 

Thanks,
Ritika C.
Selling Partner API Developer Services

**Reply by phstc** [reply]
@ritikacd, is there any concern about enabling vendor support on my existing seller app? I'm asking whether I should create a new app or combine both. Most of my customers are sellers only, but I also have vendors only, and some are both sellers and vendors. 

<img width="1235" height="722" alt="image" src="https://github.com/user-attachments/assets/dec3abd9-be3d-42dc-9596-84df9f7abfa8" />

**Reply by ritikacd** [reply]
> @ritikacd, is there any concern about enabling vendor support on my existing seller app? I'm asking whether I should create a new app or combine both. Most of my customers are sellers only, but I also have vendors only, and some are both sellers and vendors.
> 
> <img alt="image" width="1235" height="722" src="https://private-user-images.githubusercontent.com/105652/473102958-dec3abd9-be3d-42dc-9596-84df9f7abfa8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM5ODYwNzksIm5iZiI6MTc1Mzk4NTc3OSwicGF0aCI6Ii8xMDU2NTIvNDczMTAyOTU4LWRlYzNhYmQ5LWJlM2QtNDJkYy05NTk2LTg0ZGY5ZjdhYmZhOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDczMVQxODE2MTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01Y2Q1NTg1YjVhNWQ5ZTk2M2U5MDYwMGZlOWNiNGZjMTNkNTIxYWJhMDkyN2RmNGZiMDNmZTEzY2E3NzJmNGNmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.6R0FFiI2Ic6eaogaPlnLqEsRYbf1cFxtgkLblMZcXjA">

Hi @phstc, yes, you can add vendor support to the existing application. Just ensure to configure the [website authorization workflow](https://developer-docs.amazon.com/sp-api/docs/website-authorization-workflow) to have your vendors find and authorize your application.
